src/server/api/controllers/users.controller.js

```diff
@@ -44,7 +44,25 @@ const getUserFavorites = async (user_id) => {
   }
 };
 
+const editUser = async (userId, updatedUser) => {
+  if (isNaN(userId) || !userId) {
+    throw new HttpError('User Id should be a number', 400);
+  }
+
+  return knex('users').where({ id: userId }).update({
```

Comment:
I would do 3 things here:
1. Wrap this in a try/catch.
2. Co-locate error handling
3. Change what you return from this function

Reply:
Try/catch:
Currently, the code is implicit about its error handling. If the Knex call throws an error, that gets passed to the `catch` on the Promise, but it will be the database level error. It is much preferred to catch that error and instead throw a new error, like the `HttpError` you're using above. Someone who is using your API should never get to know that you are using Knex or SQL or any other internal detail. They should simply get a normal HTTP error back.

```
try {
  knex('users').where({ id: userId }).update({
    ...
  });
} catch (error) {
  throw new HttpError('Something went wrong', 500);
}

Reply:
Co-locate error handling:
Within the try/catch, I think you should check for the "missing user" case and throw a new error if that happens.

```
try {
  const updatedRows = knex('users').where({ id: userId }).update({
    ...
  });
  
  const NO_ROWS_UPDATED = 0;
  if (updatedRows === NO_ROWS_UPDATED) {
    throw new HttpError(`User with ID: ${userId} not found`, 404);
  }
  
  // ...
} 
```

Reply:
Return something meaningful:
Right now, this API will return the number `1` if all goes well because Knex reports that 1 row was updated. For a consumer of the API, what does:
`HTTP 200 1` mean? 
I think it would be better if it could respond with something like:
```
try {
  const updatedRows = knex('users').where({ id: userId }).update({
    ...
  });
  
  const NO_ROWS_UPDATED = 0;
  if (updatedRows === NO_ROWS_UPDATED) {
    throw new HttpError(`User with ID: ${userId} not found`, 404);
  }
  
  return { message: `User with ID: ${userId} updated at ${updatedAt}` }
} 
```

(Make sure to save moment().format() into a variable called `updatedAt` to use it in both places)

Reply:
I will go through them step by step. Thank you for the feedback.